docs/current/architecture.md

```diff
@@ -72,10 +72,18 @@ The `vaquero` container running in `agent` mode registers itself with an upstrea
 
 ## HA Vaquero
 
-The diagram below depicts a production deployment of Vaquero. Vaquero does not implement its own leader election and will be dependent on an outside service to manage an active-passive HA. Today we are looking for kubernetes deployments to manage the vaquero server cluster and have one running at a time. The vaquero server is backed by an etcd cluster for persistent storage of models and state. Vaquero server is essentially a message queue for its agents, the vaquero server wll never instantiate outbound calls. Vaquero agents would be deployed to service one data model "site", technically two agents could run in the same broadcast domain. We would recommend deploying two Vaquero agents per broadcast domain just for operational safety. Requirements for a production deployment would include [Docker](https://www.docker.com/), [etcd](https://github.com/coreos/etcd), and a kubernetes service to load balance between containers. See the [outage document](outage.html) to see how Vaquero handles failures.
-
 ![](jan17HA.png)
 
-## Deployment and Availability Considerations
+The diagram above depicts a production deployment of Vaquero. Vaquero implements its own leader election, and also contains a readiness probe to integrate with kubernetes deployments.
+
+#### Running vaquero with internal HA
+Vaquero implements leader election using Etcd's [built in leader election](https://godoc.org/github.com/coreos/etcd/clientv3/concurrency). Our procedure works by giving every vaquero server a unique ID. All running servers campaign on the same Etcd "leader" key, and attempt to put their own unique IDs as the "leader" value. One server will successfully become leader and serve as a fully-functional vaquero server, and the rest will continue to passively campaign. Should one of the vaquero servers exit, error out, or power off, another campaigning server will take on the role of leader within a fraction of a second.
+
+*Note*: Once a vaquero server errors out or temporarily loses contact with etcd, it is no longer permitted to continue campaigning. A failed vaquero server must be manually restarted to re-enter leader election. See the [outage document](outage.html) to see how Vaquero handles failures.
+
+Ensure you have toggled the `HA` configuration flag to `true` and that your etcd cluster is properly configured and healthy before implementing HA vaquero servers. To watch the current leader value from the command line, run `etcdctl elect -l leader`.  
```

Comment:
watching the election isn't useful, and might not always be true in the future if you set an `etcd-root`